osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisServiceImpl.java

```diff
@@ -1391,14 +1391,19 @@ private String retrieveSecretKey(String repoKey) throws Exception {
         String secretKey = null;
 
         if (repoVal != null) {
+            try {
+                // Using `repoKey` for Associated Data during decryption
+                secretKey = cipherFactory.getCipherByID(repoVal.getKeyID())
+                        .decrypt(repoVal,
+                                cipherFactory.getSecretCipherKeyByID(repoVal.getKeyID()),
+                                repoKey);
 
-            // Using `repoKey` for Associated Data during encryption
-            secretKey = cipherFactory.getCipherByID(repoVal.getKeyID())
-                    .decrypt(repoVal,
-                            cipherFactory.getSecretCipherKeyByID(repoVal.getKeyID()),
-                            repoKey);
-
-            logger.debug("[Cache] Retrieve Secret Key successful");
+                logger.debug("[Cache] Retrieve Secret Key successful");
+            } catch (Exception e) {
+                logger.error("Error: Unable to decrypt secret key data for Redis key: {}. Error details: {}", repoKey, e.getMessage());
+                logger.debug("Full stack trace:", e);
+                deleteSecretKey(repoKey);
```

Comment:
I'm not sure this is relevant but should we log something in case this fails?

Reply:
Yes we should, in fact we do not log anything for any redis operations.
I created a ticket for this tech debt: https://scality.atlassian.net/browse/S3C-8885